### ISSUE TITLE
MME Crashing when trying to connect a 5G Capable Smartphone

### ISSUE THREAD

**Issue by Marcel2508** [OPENAIRINTERFACE/openair-cn#94]
Hi, I hope someone could help me here.

I'm using the code from here: https://open-cells.com/index.php/2019/09/22/all-in-one-openairinterface/
So I'm working on Commit 724542d0b59797b010af8c5df15af7f669c1e838 from eurecom gitlab.
Could this issue already be fixed by this newer version here on github?

I'm currently trying to connect a 5G capable SmartPhone to the Setup.
At first my MME always crashed as soon as the phone connected. I tracked the error down to 
emm_data_context_get(..) returning a  Null-Pointer in emm_proc_attach_reject(..){...} 
in src/nas/emm/Attach.c

After some researching I figured out, that the parsing of the Attach Request fails, because
at one point it got the wrong IEI-Tag -> executing the default option in src/nas/emm/msg/AttachRequest.c:257

This leaves me with 2 Questions: 
1.  How could the AttachRequest Message be invalid in the first place (are all possible formats implemented, or only some?) and 
2. Why is the Method responsible for sending Attach Reject messages to the UE crashing, because emm_data_context_get returns a nullpointer (which calls hashtable_ts_get, returning KEY_NOT_EXISTS- itself) ?

Do you have some sort of documentation, where I can find the Attach-Packet's Data-Structure?
Could it be normal behaviour, that the first AttachRequest ist malformed, and a Attach Rejection is sent? Why would be there no entry in the Hashtable -> Bug somewhere else?
Or do you have any Idea, what could have caused this error?

Thanks in advance

**Comment by lionelgo** [comment]
Please try this branch: samsumg-s10-5g-g977u, and use the etc/mme.conf file.
And let us know.

**Comment by Marcel2508** [comment]
Hey, thanks for the quick reply!
I would have to update my entire Stack in order to use the "none legacy" Version of openair-cn. 
So I tried to implement 99f80e18b36fe965c3132a52ea39e3313befc6b5 myself. It did work. Now the whole connection procedure is working properly. But as soon as I enable "mobile Data" on my Device the mme crashes Again. No useful Log output is given. The issue seems to be in the following line:  https://github.com/OPENAIRINTERFACE/openair-cn/blob/724542d0b59797b010af8c5df15af7f669c1e838/src/gtpv2-c/nwgtpv2c-0.11/src/NwGtpv2c.c#L487

I found a similar issue (crashing when turning mobile data on -> https://github.com/OPENAIRINTERFACE/openair-cn/issues/45 )
But I can't find the related commit in the mentioned branch. You have any Idea what could be causing this *new* issue, or will I have to update to the latest version of openair-cn in order to get a working software-stack, compatible with my 5g Device?

Thanks again.
